[GPU] better perf model for FHS BMG kernel on some VLM shapes

skazakov1 opened the pull request:
PR addresses https://jira.devtools.intel.com/browse/MFDNN-12904 by adding Xe2/BMG kernel that is very narrowly optimized for some of VLM sizes.
Speedup on the kernel range is up to ~2x, ~30% geomean speedup on kernel's area, and ~10% on the given shape from JIRA.
No visible regression by regular perf testing.

skazakov1 commented:
make test
set test_scope=NIGHTLY
disable test_device_cpu
disable benchdnn_all
enable benchdnn_matmul
enable benchdnn_ip
enable arch_gpu_xe2-hpg-bmg
enable arch_gpu_xe2-lpg

skazakov1 commented:
make test perf-gpu
set primitive=matmul

petercad reviewed (approved):
@skazakov1 Just to check, have you run kernel generation tests in ktool on the new strategy? (`./ktool --gentest <kernel ID>`)

skazakov1 commented:
make test
set test_scope=NIGHTLY
disable test_device_cpu
disable benchdnn_all
enable benchdnn_matmul
enable benchdnn_ip
enable arch_gpu_xe2-hpg-bmg

skazakov1 commented:
make test perf-gpu
set primitive=matmul

skazakov1 commented:
> @skazakov1 Just to check, have you run kernel generation tests in ktool on the new strategy? (`./ktool --gentest <kernel ID>`)

added restriction (https://github.com/uxlfoundation/oneDNN/pull/3134), added "r" tag to the kernel, all gentests passed.

skazakov1 commented:
make test
set test_scope=NIGHTLY
disable test_device_cpu
disable benchdnn_all
enable benchdnn_matmul
enable benchdnn_ip
enable arch_gpu_xe2-hpg-bmg

skazakov1 commented:
make test perf-gpu
set primitive=matmul
